app/src/main/scala/org/alephium/explorer/util/UtxoUtil.scala

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.util
+
+import scala.collection.immutable.ArraySeq
+
+import org.alephium.explorer.api.model.{Address, Input, Output}
+import org.alephium.util.{I256, U256}
+
+object UtxoUtil {
+  def amountForAddressInInputs(address: Address, inputs: ArraySeq[Input]): U256 = {
+    inputs
+      .filter(_.address == Some(address))
+      .map(_.attoAlphAmount)
+      .collect { case Some(amount) => amount }
+      .fold(U256.Zero)(_ addUnsafe _)
+  }
+
+  def amountForAddressInOutputs(address: Address, outputs: ArraySeq[Output]): U256 = {
+    outputs
+      .filter(_.address == address)
+      .map(_.attoAlphAmount)
+      .fold(U256.Zero)(_ addUnsafe _)
+  }
+
+  def deltaAmountForAddress(address: Address,
+                            inputs: ArraySeq[Input],
+                            outputs: ArraySeq[Output]): Option[I256] = {
```

Comment:
I'm trying to remember why I choose `I256`, but I don't see any good reason :sweat_smile:

Reply:
oh of course now I remember, the `delta` can be negative, we want to show `-3.4 ALPH` if it goes out is wallet. maybe this function could return `(Option[U256], Boolean)` with the boolean precising if it's negate or not, using it to correctly represent the string (adding `-` in case of true).

Just need to change `out.sub(in)` to use the absolute value.

WDYT?

Reply:
In that case, we could use `BigInteger`. The sign of the delta cannot be ignored.

Reply:
Also, better to check the other project to see how delta is represented

Reply:
we want to represent  it the same we represent it on our explorer, which is what you can also see here for example https://www.blockchain.com/btc/address/3CVGrZH8dRirWPWgCpEHyUUJfMQnp57ir9

Reply:
I know. I thought the format used by explorer might be different from the format used for exporting. As I just checked on Etherscan, they are the same. So I am fine now.